src/pkg/packager/sources/tarball.go

```diff
@@ -204,5 +204,31 @@ func (s *TarballSource) LoadPackageMetadata(dst *layout.PackagePaths, wantSBOM b
 // Collect for the TarballSource is essentially an `mv`
 func (s *TarballSource) Collect(dir string) (string, error) {
 	dst := filepath.Join(dir, filepath.Base(s.PackageSource))
-	return dst, os.Rename(s.PackageSource, dst)
+	err := os.Rename(s.PackageSource, dst)
+	linkErr := &os.LinkError{}
+	isLinkErr := errors.As(err, &linkErr)
+	if err != nil && !isLinkErr {
+		return "", err
+	}
+	if err == nil {
+		return dst, nil
+	}
+
+	// Copy file if rename is not possible due to existing on different partitions.
+	srcFile, err := os.Open(linkErr.Old)
+	if err != nil {
+		return "", err
+	}
+	defer srcFile.Close()
+	dstFile, err := os.Create(linkErr.New)
+	if err != nil {
+		return "", err
+	}
+	defer dstFile.Close()
+	_, err = io.Copy(dstFile, srcFile)
+	if err != nil {
+		return "", err
+	}
+	return dst, nil
```

Comment:
I believe we want to delete the file from the original partition here. 

Also what are your thoughts on the "github.com/otiai10/copy".Copy() method? Is this simple enough to not be library worthy ?

Reply:
You are completely right.

I think that it could be worth discussing in a different PR, we probably need some use case and benchmarks to explain why we would pull in another dependency for this.